src/components/Teams/TeamMembersPopup.jsx

```diff
@@ -103,29 +122,40 @@ export const TeamMembersPopup = React.memo(props => {
   const icons = {
     '-1': { icon: faSortUp },
     '0': { icon: faSort, style: { color: 'lightgrey' } },
-    '1': { icon: faSortDown }
-  }
+    '1': { icon: faSortDown },
+  };
 
   const toggleOrder = useCallback(() => {
-    setSortOrder((pre) => {
+    setSortOrder(pre => {
       if (pre !== -1) {
         return pre - 1;
       }
       return 1;
-    })
-  }, [])
+    });
+  });
+
+  // call the handler to update the team member's visibility
+  const UpdateTeamMembersVisiblity = (userId, choice) => {
+    props.onUpdateTeamMemberVisiblity(userId, choice);
+  };
 
   useEffect(() => {
-    sortList(sortOrder)
-  }, [props.members.teamMembers, sortOrder])
+    sortList(sortOrder);
+    setTeamVisibility(props.teamData);
+  }, [props.members.teamMembers, sortOrder]);
 
   useEffect(() => {
     onValidation(true);
   }, [props.open]);
 
+  useEffect(() => {
+    setTeamVisibility(props.teamData);
+  }, [props.teamData]);
```

Comment:
Nitpick: add a line break after this `useEffect()` to improve code readability and consistency of spacing.